src/routes/compiler/compiler.ts

```diff
@@ -1,20 +1,25 @@
 import solc from 'solc';
 
-import type { ContractSources } from "$lib/models/solc";
-import type { CompilerInput } from "$lib/models/solc";
+import type { CompiledContract, CompilerInput , ImportContents} from "$lib/models/solc";
```

Comment:
Are the changes in this file functionally different from what it was before?  Otherwise, I would prefer to keep the previous code since it is more direct.  Using many small functions makes it harder to keep track of what is happening.  There is also an unnecessary series of `JSON.parse` and `JSON.stringify`.